### ISSUE TITLE
STRtree.query() segfaults when run in multiple processes

### ISSUE THREAD

**Issue by peterbriggs42** [Toblerity/Shapely#855]
## Expected behavior and actual behavior.
If you spin up multiple processes using the `multiprocessing` standard library module and create an STRtree for a collection of polygons, and then query that STRtree, then one of the child processes will segfault. 

I would expect that the STRtree object would be process-safe, i.e. a child process can create and query its own STRtree without worrying about other processes.

## Steps to reproduce the problem.

```
import multiprocessing
from shapely.geometry import Point, Polygon
from shapely.strtree import STRtree

def run():
    polys = []
    for i in range(1000):
        triangle = Polygon([(i, i), (i+1, i+1), (i+2, i), (i, i)])
        polys.append(triangle)
    tree = STRtree(polys)
    for p in polys:
        point = Point(p.exterior.coords[0])
        result = tree.query(point) # this line will segfault

processes = []
for i in range(2):
    p = multiprocessing.Process(target=run)
    processes.append(p)
    p.start()
```

## Operating system
CentOS 7

## Shapely version and provenance
Shapely 1.7.0, installed from pip

**Comment by sgillies** [comment]
@peterbriggs42 thank you for the script! Indeed, I also expect this to be safe, and I have had no success reproducing the problem with the shapely wheels on my macbook (python 3.6.8, OS X 10.13.6) or thinkpad (python 3.6.7, Ubuntu 18.04).

Can you provide more information? Does the program crash if you have one child process (1 instead of 2). Does it crash without multiprocessing?

**Comment by peterbriggs42** [comment]
On the CentOS VMs (7.7.1908 and 7.6.1810) I have been using to reproduce the issue, the segfaults are directly caused by the use of multiple processes. In a single-process context (or if you use `multiprocessing` to spin up a single child process), the script runs just fine. If you use multiple child processes, all but one of the child processes will segfault and be killed by the OS.

I spun up a fresh Ubuntu VM (also 18.04) and did not see a segfault. One of my colleagues tried it out on his MacOS laptop and was also unable to reproduce the segfault, so I wonder if it is tied somehow to the CentOS operating system.

**Comment by sgillies** [comment]
@peterbriggs42 the shapely manylinux1 wheels are built using a CentOS 5 based image, if that gives you any clue. I would expect your program to succeed on your VM if you install the geos-devel package and build shapely from a source distribution like `pip install --no-binary shapely shapely`. It might be worth checking to see if that's a viable workaround. That might be the wrong word, workaround, because the really solid way to deploy shapely is like that, the wheels are mostly about making it easy to start using shapely.

**Comment by peterbriggs42** [comment]
I built GEOS 3.6.4 (and also the latest version, 3.9.0) from source against a fresh install of CentOS 7. I installed numpy and Cython, and then did a `pip install --no-binary shapely shapely`, but in all cases with CentOS I'm still getting segfaults on my CentOS VM. I also tried the above with Shapely v1.6.4, retrieved via pip with --no-binary.

I also tried installing GEOS via the `geos-devel` package from yum, but that yielded the same results.
